static/js/publisher/form/fileUpload.js

```diff
@@ -0,0 +1,82 @@
+import React from "react";
+
+import PropTypes from "prop-types";
+
+import { validateRestrictions } from "../../libs/fileValidation";
+
+class FileUpload extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.addFileHandler = this.addFileHandler.bind(this);
+    this.fileChangedHandler = this.fileChangedHandler.bind(this);
+    this.keyboardEventHandler = this.keyboardEventHandler.bind(this);
+  }
+
+  addFileHandler() {
+    const { restrictions, inputName, parent } = this.props;
+    const input = document.createElement("input");
+    input.type = "file";
+    if (restrictions.accept) {
+      input.accept = restrictions.accept.join(",");
+    }
+    input.name = inputName;
+    input.hidden = "hidden";
+    input.addEventListener("change", () => {
+      this.fileChangedHandler(input);
+    });
+
+    if (parent && parent.holder) {
+      parent.holder.appendChild(input);
+    } else {
+      this.holder.appendChild(input);
+    }
+
+    input.dispatchEvent(new Event("click", { bubbles: false }));
+  }
+
+  fileChangedHandler(input) {
+    const { fileChangedCallback, restrictions } = this.props;
+    const files = Array.from(input.files).map(file =>
+      validateRestrictions(file, restrictions)
+    );
+
+    Promise.all(files).then(values => {
+      fileChangedCallback(values);
+    });
+  }
+
+  keyboardEventHandler(e) {
+    if (e.key === "Enter") {
+      this.addFileHandler();
+    }
+  }
+
+  render() {
+    const { classes, children } = this.props;
```

Comment:
I think that for consistency with native `className` it would be nicer to have `className` prop on component instead of custom array `classes`. So regardless of what it is just using `className` would work.